pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -769,6 +769,9 @@ func (c *Reconciler) runNextSchedulableTask(ctx context.Context, pr *v1.Pipeline
 			continue
 		}
 
+		// propagate previous task results
+		resources.PropagateResults(rpt, pipelineRunFacts.State)
+
```

Comment:
please add a test in [pipelinerun_tests.go ](https://github.com/tektoncd/pipeline/blob/main/pkg/reconciler/pipelinerun/pipelinerun_test.go)

Reply:
`PropagateResults` have been tested in resources, since the logic of `resources.PropagateResults(rpt, pipelineRunFacts.State)` is in the `runNextSchedulableTask` function, no additional test is added in pipelinerun_test.go

Reply:
but if someone removed this line, there won't be any failing tests, right? 

the test would be to ensure that the results are actually propagated during reconciliation

also reminds me that we should add an e2e test, similar to: https://github.com/tektoncd/pipeline/blob/main/test/propagated_params_test.go